pkg/xds/server/components.go

```diff
@@ -174,8 +188,13 @@ func DefaultDataplaneSyncTracker(rt core_runtime.Runtime, reconciler, ingressRec
 	if err := rt.Metrics().Register(xdsGenerationsErrors); err != nil {
 		return nil, err
 	}
+	meshSnapshotCache := mesh.NewCache(rt.ReadOnlyResourceManager(),
+		rt.Config().Store.Cache.ExpirationTime, meshResources, rt.LookupIP())
+	claCache := cla.NewCache(rt.ReadOnlyResourceManager(), rt.Config().Multicluster.Remote.Zone,
+		rt.Config().Store.Cache.ExpirationTime, rt.LookupIP())
 	return xds_sync.NewDataplaneSyncTracker(func(key core_model.ResourceKey, streamId int64) util_watchdog.Watchdog {
 		log := xdsServerLog.WithName("dataplane-sync-watchdog").WithValues("dataplaneKey", key)
+		prevHash := ""
```

Comment:
I'm wondering. Are we reusing the same underlying slice of bytes that string for every prevHash? I don't want to have copy of this huge string for every goroutine

Reply:
I made a small example, I think it emulates the behaviour of Kuma CP and `prevHash`:
```go
func stringptr(s string) uintptr {
	return (*reflect.StringHeader)(unsafe.Pointer(&s)).Data
}

func main() {
	m := map[string]string{
		"key": generateBigString(),
	}

	get := func() string {
		return m["key"]
	}
	mtx := sync.Mutex{}
	wg := sync.WaitGroup{}
	for i := 0; i < 100; i++ {
		wg.Add(1)
		go func(idx int) {
			mtx.Lock()
			defer mtx.Unlock()
			defer wg.Done()

			str := get()
			fmt.Println(idx, stringptr(str))
		}(i)
	}
	wg.Wait()
}
```
All pointers are the same, so I think yes, all goroutines are using the same underlying slice of bytes.

Reply:
great, thanks for the research